docs/source/notes/distributed_autograd.rst

```diff
@@ -307,7 +315,11 @@ The :class:`~torch.distributed.optim.DistributedOptimizer` operates as follows:
    each of the worker nodes and holds an ``RRef`` to them.
 4. When :meth:`torch.distributed.optim.DistributedOptimizer.step` is invoked,
    the distributed optimizer uses RPC to remotely execute all the local
-   optimizers on the appropriate remote workers.
+   optimizers on the appropriate remote workers. A distributed autograd 
+   `context_id` must be provided as input to 
```

Comment:
Would I be correct if I assume we used to infer the id, but now we no longer support that?

Reply:
Yes, the context_id is required now: https://github.com/pytorch/pytorch/blob/master/torch/distributed/optim/optimizer.py#L125

This was part of the change in https://github.com/pytorch/pytorch/issues/33480

Reply:
@mrshenli Here is the closed PR for this change: #33711. We used to the depend on the thread_local `context_id` in the autograd container before, and internally call `currentContext`. Now the context_id is explicitly passed in and we internally call `retrieveContext(context_id)`. dist_optim.step() also takes in the context_id now.